typescript/src/client/policies/policies.ts

```diff
@@ -0,0 +1,246 @@
+import {
+  PoliciesClientInterface,
+  CreatePolicyOptions,
+  ListPoliciesOptions,
+  ListPoliciesResult,
+  GetPolicyByIdOptions,
+  DeletePolicyOptions,
+  UpdatePolicyOptions,
+} from "./policies.types.js";
+import { CdpOpenApiClient } from "../../openapi-client/index.js";
+import { CreatePolicyBodySchema, UpdatePolicyBodySchema } from "../../policies/schema.js";
+import { Policy } from "../../policies/types.js";
+
+/**
+ * The namespace containing all Policy methods.
+ */
+export class PoliciesClient implements PoliciesClientInterface {
+  /**
+   * Lists policies belonging to the developer's CDP Project.
+   * Can be filtered by scope (project or account).
+   *
+   * @param {ListPoliciesOptions} [options] - Options for filtering and paginating the results
+   * @param {string} [options.scope] - Filter policies by scope ('project' or 'account')
+   * @param {number} [options.pageSize] - Maximum number of policies to return
+   * @param {string} [options.pageToken] - Pagination cursor for fetching next page of results
+   *
+   * @returns {Promise<ListPoliciesResult>} A paginated list of policies
+   *
+   * @example **List all policies**
+   *          ```ts
+   *          const { policies } = await cdp.policies.listPolicies();
+   *          ```
+   *
+   * @example **Filter by scope**
+   *          ```ts
+   *          const { policies } = await cdp.policies.listPolicies({
+   *            scope: 'project'
+   *          });
+   *          ```
+   *
+   * @example **With pagination**
+   *          ```ts
+   *          // Get first page
+   *          const firstPage = await cdp.policies.listPolicies({
+   *            pageSize: 10
+   *          });
+   *
+   *          // Get next page using cursor
+   *          const nextPage = await cdp.policies.listPolicies({
+   *            pageSize: 10,
+   *            pageToken: firstPage.pageToken
+   *          });
+   *          ```
+   */
+  async listPolicies(options: ListPoliciesOptions = {}): Promise<ListPoliciesResult> {
+    return CdpOpenApiClient.listPolicies(options) as Promise<ListPoliciesResult>;
```

Comment:
The `as`'s are used because the zod schemas/types properly discriminate rule criteria unions by operation. In the openapi types, it's possible to have `ethValue` criteria under a `signSolTransaction` operation, but impossible with the zod types, which is more correct.